### ISSUE TITLE
Missing `readonly` and `declare` in `ClassMethod` and `TSMethodSignature`

### ISSUE THREAD

**Issue by fisker** [prettier/prettier#9760]
**Prettier 2.2.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgMI7AA6UOOATnOgCbSoCeOAVHABQCUhAviWdXGAyVmAMw7cSPKCQCWsOORHowcHADEIEQrxz9B6YW3YBuHegBGmGOWUwcYk5JAAaEBAAOMGdEzJQB8ggAdwAFAwRfFHRUIPR6X1dzGzAAazgYAGV3ZTkAc2RrAFc4VzgAW3M4an5qABl0KFzC9Fy4DXIy9BgvRuQQdEKYCBcQAAsYMtQAdVGZeExslQyIuZkANzn6PrBsEblMBRgQm1zO5CVUA9cAK0wADwAhZLTM9DK4Wrk4c+irkFu7hk8qg4ABFQoQeA-S4lEDZcgHch9GD0dxwTBgcgyTwjdxY2BTGTUGCjZAADgADK48RADlMbO4+nj0Qo1t9XABHCHwY4eSL9TAAWigcCqVRGlC5MkoxxaZyQFz+BzKMgK5GKrkwwLB3O+Ct+sJgFkJxNJSAATK5rOgZKg8ngIGV5SB0QBWEaFA4AFQskUVsLWxQAklB+LAMpjsTAAIKhjIokHQg5cLhAA)

```sh
--parser typescript
```

**Input:**

```tsx
class C {
  readonly *e() {}
  declare *f() {}
}

interface Foo {
  declare e();
  abstract f();
}
```

**Output:**

```tsx
class C {
  *e() {}
  *f() {}
}

interface Foo {
  e();
  f();
}

```

**Expected behavior:**
```tsx
class C {
  readonly *e() {}
  declare *f() {}
}

interface Foo {
  declare e();
  abstract f();
}
```

---

Update 

**Expected behavior:**

Throw SyntaxError.

**Comment by brodybits** [comment]
> [Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgMI7AA6UOOATnOgCbSoCeOAVHABQCUhAviWdXGAyVmAMw7cSXEABoQEAA4wAltEzJQ6cuQgB3AAqaEalOlQ709NbIBG5dGADWcGAGV59pVADmyGOQCucLJwALbWcNT81AAy6N7+6F5wAGIQ5CHoMMreyCDo-jAQMiAAFjAhqADqJUrwmO5gcC5GtUoAbrX0uWDYxZ6YcOQwenZeGcgipgOyAFaYAB4AQnaOzi7oIXDRnnATU0Egc-Munl6ocACK-hDwe6jTIO7kA+S5MPTycJhg5EqKxfJfrBKkpqDASsgABwABlkgIgA0qdnkuUBX0GbV2sgAjtd4CMFMY8pgALRQOARCLFSi4pSUEaJcZISb3A4DEJKXwBNmnc5XG67Zn7WQwdDWEFgiFIABMIrsSlQpzwEBCTJAXwArMV-AMACpi4wsh5tQIASSg-FgLh+fxgAEELS53uc7gMuFwgA)

Same output comes from both `typescript` and `babel-ts` parsers (other JavaScript parsers signal a syntax error).

Background is in PR #9748 & <https://github.com/typescript-eslint/typescript-eslint/issues/2908>.

I gather that the testing with AST_COMPARE=1 does not show an issue because the attributes are not present in the AST coming either from` babel-ts` or `typescript` parser.

**Comment by fisker** [comment]
> I gather that the testing with AST_COMPARE=1 does not show an issue because the attributes are not present in the AST 

 https://github.com/prettier/prettier/pull/9768

**Comment by fisker** [comment]
Invalid `declare` on accessors not throwing, should be the same problem. `babel-ts` fixed this in #10750.

**Prettier 2.2.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgGIQQ7AA6UOOAJnGBgE5w4DmcMOAZoQBQCUZF1WugY5MrDtwBufKAF8QAGhAQADjACW0TMlDC6EAO4AFYQm0p0qA+gCe2pQCM66MAGtWAZRUv1UJshg6AFc4JTgAWwc4SmpKABl0PyD0FgI6cPQYDT9kEHQgmAhFEAALGHDUAHUS9XhMbzA4DzNa9Ulam1ywbGLfMToYI2cmDOR2SzElACtMAA8AIWc3T3RwuDjfODGJ0JAZ2Y9fJlQ4AEUgiHht1EmQbzp+3JgbFThMMDp1NWKVT9hK9SUGAlZAADgADEpfhAxJVnCpcr83nA6JItkoAI4XeBDVTmPKYAC0UDg0WixQYWPUDCGKVGSHGN12YnC6gCwWZRxO50uWwZOyUMHQDgBQJBSAATILnOpUEcAMIQcL0kBvACsxSCYgAKsLzIzbpIQgBJKDUWAeD5fGAAQTNHmeJ2uYlksiAA)

```sh
--parser typescript
```

**Input:**

```tsx
class Foo {
  declare get foo()
  declare set foo(v)
}
```

**Output:**

```tsx
class Foo {
  get foo();
  set foo(v);
}

```
